catch/unit/texture/hipTextureObjectTests.cc

```diff
@@ -0,0 +1,220 @@
+  hipArray* array_member;
+  size_t size; /* size in bytes*/
+  int width;   /* width in elements */
+
+  TextureObjectTestWrapper(bool useResourceViewDescriptor, bool ommitDestroy = false)
+      : ommit_destroy_(ommitDestroy), width(128) {
+    int i;
+    size = width * sizeof(float);
+
+    host_data_ = (float*)malloc(size);
+    memset(host_data_, 0, size);
+
+    for (i = 0; i < width; i++) {
+      host_data_[i] = i;
+    }
+
+    channel_desc = hipCreateChannelDesc(32, 0, 0, 0, hipChannelFormatKindFloat);
+    hipMallocArray(&array_member, &channel_desc, width);
+
+    HIP_CHECK(
+        hipMemcpy2DToArray(array_member, 0, 0, host_data_, size, size, 1, hipMemcpyHostToDevice));
+
+    memset(&res_desc, 0, sizeof(res_desc));
+    res_desc.resType = hipResourceTypeArray;
+    res_desc.res.array.array = array_member;
+
+    memset(&tex_desc, 0, sizeof(tex_desc));
+    tex_desc.addressMode[0] = hipAddressModeClamp;
+    tex_desc.filterMode = hipFilterModePoint;
+    tex_desc.readMode = hipReadModeElementType;
+    tex_desc.normalizedCoords = false;
+
+    memset(&res_vew_desc, 0, sizeof(res_vew_desc));
+
+    if (useResourceViewDescriptor) {
+#if HT_AMD
+      res_vew_desc.format = hipResViewFormatFloat1;
+      res_vew_desc.width = size;
+#else
+      std::cout << "Resource View Descriptors are not supported on NVIDIA currently" << std::endl;
+      useResourceViewDescriptor = false;
+#endif
+    }
+
+
+    HIP_CHECK(hipCreateTextureObject(&texture_object, &res_desc, &tex_desc,
+                                     useResourceViewDescriptor ? &res_vew_desc : nullptr));
+  }
+
+  ~TextureObjectTestWrapper() {
+    if (!ommit_destroy_) {
+      HIP_CHECK(hipDestroyTextureObject(texture_object));
+    }
+    HIP_CHECK(hipFreeArray(array_member));
+    free(host_data_);
+  }
+};
+
+/* hipGetTextureObjectResourceDesc tests */
+
+TEST_CASE("Unit_hipGetTextureObjectResourceDesc_positive") {
+  CHECK_IMAGE_SUPPORT;
+
+  TextureObjectTestWrapper tex_obj_wrapper(false);
+
+  hipResourceDesc check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  HIP_CHECK(hipGetTextureObjectResourceDesc(&check_desc, tex_obj_wrapper.texture_object));
+
+  REQUIRE(check_desc.resType == tex_obj_wrapper.res_desc.resType);
+  REQUIRE(check_desc.res.array.array == tex_obj_wrapper.res_desc.res.array.array);
+}
+
+
+TEST_CASE("Unit_hipGetTextureObjectResourceDesc_Negative_Parameters") {
+  CHECK_IMAGE_SUPPORT;
+
+  TextureObjectTestWrapper tex_obj_wrapper(false);
+
+  hipResourceDesc check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  SECTION("desc is nullptr") {
+    HIP_CHECK_ERROR(hipGetTextureObjectResourceDesc(nullptr, tex_obj_wrapper.texture_object),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("texture is invalid") {
+    HIP_CHECK_ERROR(
+        hipGetTextureObjectResourceDesc(&check_desc, static_cast<hipTextureObject_t>(0)),
+        hipErrorInvalidValue);
+  }
+}
+
+/* hipGetTextureObjectResourceViewDesc tests */
+
+#if HT_AMD
```

Comment:
same

Reply:
@chrispaquot   Unfortunately, it is not possible as hipGetTextureObjectResourceViewDesc does not compile on Nvidia.